### PR TITLE
support customising GraphQL fields from console

### PR DESCRIPTION
**Pull request by wawhal**
Fixes internal issue 270 

### Description

You can now edit the root fields associated with any table from the table's `Modify` page.
You can also edit the GraphQL fields associated with the columns of tables. 

### Affected components 

- Console

### CODE REVIEW

File: console/src/components/Services/Data/Common/ReusableComponents/RootFieldEditor.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          placeholder={`${getDefaultRootField(rfType)} (default)`}
+          className="form-control"
+          onChange={onChange}
+          disabled={disabled}
+        />
+      </div>
+    </div>
+  );
+
+  const getQuerySection = () => {
```
Comment (karthikvt26) [inline console/src/components/Services/Data/Common/ReusableComponents/RootFieldEditor.js]: Will it add value if we are able to refactor this functionality into a seperate function and consume it by args (for query and mutation)?

File: console/src/components/Services/Data/TableModify/ColumnEditor.js [hunk 2]
```diff
@@ -71,6 +75,25 @@ const ColumnEditor = ({
     dispatch(editColumn(colName, 'isUnique', e.target.value === 'true'));
   };
 
+  const getColumnAliasInput = () => {
+    if (!supportAliasing) return;
+    return (
+      <div className={`${styles.display_flex} form-group`}>
+        <label className="col-xs-2">Custom GraphQL field</label>
```
Comment (karthikvt26) [inline console/src/components/Services/Data/TableModify/ColumnEditor.js]: This seems little weird as other columns are one liner and this extends to 3 lines

File: console/src/components/Services/Data/TableModify/ModifyActions.js [hunk 3]
```diff
@@ -1520,26 +1592,61 @@ const saveColumnChangesSql = (colName, column, onSuccess) => {
...
+    /* column alias up/down migration*/
+    if (SUPPORT_GRAPHQL_ALIASING) {
+      const existingCustomColumnNames = getTableCustomColumnNames(table);
+      const existingRootFields = getTableCustomRootFields(table);
+      const newCustomColumnNames = { ...existingCustomColumnNames };
+      let isAliasChanged = false;
+      if (alias) {
+        if (alias !== existingCustomColumnNames[alias]) {
+          isAliasChanged = true;
+          newCustomColumnNames[colName] = alias;
```
Comment (karthikvt26) [inline console/src/components/Services/Data/TableModify/ModifyActions.js]: Sanitise column alias too?

File: console/src/components/Services/Data/Common/ReusableComponents/RootFieldEditor.js (hunk 1 above)
Comment (wawhal) [reply console/src/components/Services/Data/Common/ReusableComponents/RootFieldEditor.js]: Yeah. Reduces some repeated code. Changed.

File: console/src/components/Services/Data/TableModify/ColumnEditor.js (hunk 2 above)
Comment (wawhal) [reply console/src/components/Services/Data/TableModify/ColumnEditor.js]: Fixed.

**Comment by arvi3411301** [comment]
@rikinsk CLI change is not required for the version parameter.

File: console/src/components/Services/Data/TableModify/RootFields.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      [field]: alias,
+    };
+    dispatch(modifyRootFields(newRootFields));
+  };
+
+  const note =
+    'You can customize the GraphQL root fields associated with this table.';
+
+  const rootFields = (aliases, disabled) => {
+    return (
```
Comment (karthikvt26) [inline console/src/components/Services/Data/TableModify/RootFields.js]: Lets change this behaviour for disabled as currently onChange functions will be sent while component is disabled. Or is it required? if not 

we can have a JSON object 
```
const onChangeHandlers = {};
if (!disabled) {
    // Attach handlers to specific keys
} 

// Rest

<RootFieldEditor
   ... rest
   { ...onChangeHandlers}
/>
```

**Comment by altitudems** [comment]
Can't wait to use this feature!

**Comment by weyert** [comment]
Can't wait to use this feature! Looks really useful!

**Pull request merged**